action.yml

```diff
@@ -75,14 +77,56 @@ runs:
         if [ "$VALUES_FILE" ]; then
           cp "$VALUES_FILE" /tmp/values.yaml
         else
-          touch /tmp/values.yaml          
+          touch /tmp/values.yaml
+        fi
+
+        echo "$TIMEOUT" | grep -Eq '^[0-9]+(s|m)$' || (
+          echo "ERROR: Invalid timeout value provided: $TIMEOUT" >&2
+          echo "       Supported format: [0-9]+(s|m)" >&2
+          false
+        )
+        export TIMEOUT_SECONDS="$(echo "$TIMEOUT" | awk '{print match($1, "^([0-9]+)m$", m) ? substr($1, RSTART, RLENGTH-1)*60 : gensub("s$", "", "1", $1)}' -)"
```

Comment:
This doesn't work if the user sets the `TIMEOUT` to `5m10s`, a format which `helm` supports:

```
      --timeout duration                           time to wait for any individual Kubernetes operation (like Jobs for hooks) (default 5m0s)
```